frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx

```diff
@@ -73,60 +78,49 @@ const ManageStorageModal: React.FC<AddStorageModalProps> = ({ existingData, isOp
 
     const pvc = assemblePvc(name, namespace, description, size);
 
-    const handleError = (e: Error) => {
-      setError(e);
-      setActionInProgress(false);
-    };
-    const handleNotebookNameConnection = (pvcName: string) => {
-      if (notebookName) {
-        attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value)
-          .then(() => {
-            setActionInProgress(false);
-            onBeforeClose(true);
-          })
-          .catch((e) => {
-            setError(e);
-            setActionInProgress(false);
-          });
-      } else {
-        setActionInProgress(false);
-        onBeforeClose(true);
-      }
-    };
-
     if (existingData) {
       const pvcName = existingData.metadata.name;
+      const pvcPromises: Promise<PersistentVolumeClaimKind | NotebookKind>[] = [];
       if (getPvcDisplayName(existingData) !== createData.nameDesc.name) {
-        await updatePvcDisplayName(pvcName, namespace, createData.nameDesc.name);
+        pvcPromises.push(updatePvcDisplayName(pvcName, namespace, createData.nameDesc.name));
       }
       if (getPvcDescription(existingData) !== createData.nameDesc.description) {
-        await updatePvcDescription(pvcName, namespace, createData.nameDesc.description);
+        pvcPromises.push(updatePvcDescription(pvcName, namespace, createData.nameDesc.description));
+      }
+      if (parseInt(getPvcTotalSize(existingData)) !== createData.size) {
+        pvcPromises.push(updatePvcSize(pvcName, namespace, `${createData.size}Gi`));
       }
       if (removedNotebooks.length > 0) {
         // Remove connected pvcs
-        Promise.all(
-          removedNotebooks.map((notebookName) =>
+        pvcPromises.push(
+          ...removedNotebooks.map((notebookName) =>
             removeNotebookPVC(notebookName, namespace, pvcName),
           ),
-        )
-          .then(() => handleNotebookNameConnection(pvcName))
-          .catch(handleError);
-        return;
-      }
-      handleNotebookNameConnection(pvcName);
-      if (parseInt(getPvcTotalSize(existingData)) !== createData.size) {
-        await updatePvcSize(pvcName, namespace, `${createData.size}Gi`);
+        );
       }
+      Promise.all(pvcPromises)
+        .then(() => {
+          if (notebookName) {
+            pvcPromises.push(attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value));
```

Comment:
Shouldn't you return the promise from `attachNotebookPVC` ?
```suggestion
            return attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value);
```